src/dune_rules/context.ml

```diff
@@ -516,7 +516,12 @@ let create ~(kind : Kind.t) ~path ~env ~env_nodes ~name ~merlin ~targets
     let env =
       let cwd = Sys.getcwd () in
       let extend_var var ?(path_sep = Bin.path_sep) v =
-        let v = Filename.concat cwd (Path.Build.to_string v) in
```

Comment:
What about just using `Path.Build.to_absolute_filename`?

Reply:
That doesn't seem to exist. Let me see if I can implement it for `Path.Build`.

Reply:
It's just a matter of `Path.to_absolute_filename (Path.build v))`

Reply:
I think there's a bit of a bootstrapping issue with that. My implementation:

```ocaml
  let to_absolute_filename (t : t) =
    Outside_build_dir.to_absolute_filename
      (Outside_build_dir.append_local (Fdecl.get build_dir) t)
```

Reply:
Okay, but you should change `Path.to_absolute_filename` to use your implementation. Otherwise, it's weird that `Path.to_absolute_filename (Path.build v) <> Path.Build.to_absolute_filename v`

Reply:
I think they do the same thing already:

```ocaml
Path.to_absolute_filename (Path.build v) 

=> Path.to_absolute_filename (In_build_dir s)

=> Outside_build_dir.to_absolute_filename (Outside_build_dir.append_local (Fdecl.get Build.build_dir) s)
```

Which is exactly my implementation

Reply:
Hmm, then what's the "bootstrapping issue" then?

In any case, one of them should call the other to avoid the duplicate implementation.

Reply:
the bootstrapping issue is that `module T` (which defines `In_build_dir` and such) depends on `Path.Build`. So we can't get a `In_build_dir` from a `Path.Build.t`.

Reply:
Oh, what i meant was not to introduce `Path.Bath.to_absolute_filename` at all. and just call `Path.to_absolute_filename (Path.build v)` directly in `context.ml`.

Reply:
Fixed that and added a changes entry